pnc/src/main/java/org/jboss/pnc/bacon/pnc/client/BifrostClient.java

```diff
@@ -31,9 +37,29 @@ public BifrostClient(URI baseUrl) {
         client = HttpClients.createDefault();
     }
 
-    public void writeLog(String id, boolean follow, Consumer<String> onLine) throws IOException {
-        String query = "direction=ASC" + "&matchFilters=mdc.processContext.keyword:build-" + id
-                + "&prefixFilters=loggerName.keyword:org.jboss.pnc._userlog_";
+    public List<String> getLog(String buildId, LogType logType) throws IOException {
+        List<String> logs = new ArrayList<>();
+        writeLog(buildId, false, logs::add, logType);
+        return logs;
+    }
+
+    public void writeLog(String id, boolean follow, Consumer<String> onLine, LogType logType) throws IOException {
+
+        String query;
+
+        switch (logType) {
+            case BUILD:
+                query = "direction=ASC" + "&matchFilters=mdc.processContext.keyword:build-" + id
+                        + "&prefixFilters=loggerName.keyword:org.jboss.pnc._userlog_.build-log";
+                break;
+            case ALIGNMENT:
```

Comment:
```
MessageFormat.format("direction=ASC&matchFilters=mdc.processContext:build-{0},loggerName:org.jboss.pnc._userlog_.alignment-log",id)
```
Cleaner code and maybe more optimized by moving loggerName to matchFilter. 

For BUILD and ALIGNMENT, to increase performance, add smthing alongside `&batchSize=5000` and `&batchDelay=500`. 

Plus, there are additional formats to log lines. You can configure with `&format=XXXX`. Formats are TIMESTAMP (timestamp + message), PLAIN (just message), LEVEL (logLevel + message), DEFAULT (timestamp + loggerName + message). 

Are you using `/text` or `/` endpoint?